### DISCUSSION TITLE
What SVG filter elements are supported?

### DISCUSSION THREAD

**Discussion by victorjkhalaf** [libvips/libvips#3220] [Q&A] [answered]
We are trying to move away from using svg-to-jpeg headless Chromium-based converters and found libvips could get us a lot of what we are looking for with its native JPG `jpegsave`. It works great for images embedded into an SVG with some scale and transforms applied.
Unfortunately one area it seems to not entirely work is when filter primitive elements (https://developer.mozilla.org/en-US/docs/Web/SVG/Element#filter_primitive_elements) are applied in an SVG. Is there a list of what is and isn't supported anywhere?

**Comment by jcupitt** [comment, marked as answer]
Hi @victorjkhalaf,

libvips uses librsvg for rendering:

https://gitlab.gnome.org/GNOME/librsvg

So you'd need to check the feature list there.

libresvg have a handy feature table:

https://github.com/W3SS/libresvg/blob/master/doc/support-table.adoc

**Reply by lovell** [reply]
This feature table link appears to be to a fork last updated 5 years ago.

The latest version of librsvg supports almost all level 1 filter effects to some degree - see https://gitlab.gnome.org/GNOME/librsvg/-/blob/main/src/element.rs#L680

**Reply by jcupitt** [reply]
Oh haha I didn't check the date. Ahem, sorry librsvg.

**Comment by victorjkhalaf** [comment]
Many thanks @jcupitt even if for now libvips won't meet the svg needs (it does for most of our other bitmap image manipulations), this is good clarity for us. :)

**Reply by jcupitt** [reply]
It looks like libresvg might do more of what you need. There's been talk of adding libresvg support to libvips, but no one's got around to it :(

**Reply by victorjkhalaf** [reply]
We actually dove deeper and found that we had set something that was not in spec in one of the filter elements used. Now whatever is supported so far in libvips, works for us!
We can close this discussion. :)
